### PR TITLE
test: #2534 Replace use of TestExporter with InMemoryExporter

### LINKED ISSUE #2534: Consider removing TestExporter
`InMemoryExporter` is designed for unit test purpose.

If OpenTelemetry .NET unit tests have some common need that cannot be achieved by `InMemoryExporter`, we would want to fix it in `InMemoryExporter` rather than inventing another exporter for tests.

The ask is to evaluate whether it is possible to get rid of https://github.com/open-telemetry/opentelemetry-dotnet/blob/0c93744d4ba51676b61ab1be38f7764190cfcec0/test/OpenTelemetry.Tests/Shared/TestExporter.cs and use `InMemoryExporter`.

### PR DESCRIPTION
**Pull request by chertby**
Related to #2534.

Replace use of TestExporter with InMemoryExporter in OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs [hunk 1]
```diff
@@ -57,7 +57,8 @@ public void ToOtlpResourceMetricsTest(bool includeServiceNameInResource)
 
             using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{includeServiceNameInResource}", "0.0.1");
 
-            var metricReader = new BaseExportingMetricReader(new TestExporter<Metric>(RunTest))
+            var exportedItems = new List<Metric>();
+            var metricReader = new BaseExportingMetricReader(new InMemoryExporter<Metric>(exportedItems))
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs:61]: Suggestion:
1. pass this directly to the `AddReader`, no need to create `metricReader` variable.
2. change `metricReader.Collect()` to `provider.ForceFlush()`.

Similar code can be found from https://github.com/open-telemetry/opentelemetry-dotnet/blob/5325185d0221e363e30c7daa45610d01f8bfb451/test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs#L43.

**Review by reyang — APPROVED** [review]
LGTM.

**Comment by reyang** [comment]
> Fixes #2534.

I've changed the description to "Related to #2534." since it doesn't fix the entire issue (the goal is that there is no `TestExporter`).

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs (hunk 1 above)
Comment (chertby) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: done, thnx

Comment (reyang) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs]: Thanks! 👍

**Review by cijothomas — APPROVED** [review]

**Pull request merged**